optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/steps/schema/DeleteIndexTemplateIfExistsStep.java

```diff
@@ -31,24 +32,22 @@ public DeleteIndexTemplateIfExistsStep(final String templateName, final int temp
     this.templateVersion = templateVersion;
   }
 
-  @Override
-  public UpgradeStepType getType() {
-    return SCHEMA_DELETE_TEMPLATE;
-  }
-
   @Override
   public IndexMappingCreator getIndex() {
     throw new UpgradeRuntimeException(
         "Index class does not exist as its template is being deleted");
   }
 
+  @Override
+  public UpgradeStepType getType() {
+    return SCHEMA_DELETE_TEMPLATE;
+  }
+
   @Override
   public void execute(final SchemaUpgradeClient schemaUpgradeClient) {
     final String indexAlias =
         schemaUpgradeClient.getIndexNameService().getOptimizeIndexAliasForIndex(templateName);
-    schemaUpgradeClient.getAliasMap(indexAlias).keySet().stream()
-        .filter(templateName -> templateName.contains(this.templateName))
-        .forEach(schemaUpgradeClient::deleteTemplateIfExists);
```

Comment:
I think there was a bug here, because the alias would already have been deleted at this point in time, so depending on aliases to find the template will not work. It was not detected because the assertion in the IT was also detected on the wrong version 🙈